### PR TITLE
Split chunks into individual dirs

### LINKED ISSUE #4203: 32MB in mermaid@10.0.2/dist/ with duplicate files?
### Description

When I port  Rust-aquamarine from mermaid 9.2 to to mermaid 10.0.2 (https://github.com/mersinvald/aquamarine/issues/35), it seems the JS footprint has grown from 900KB to 32MB. I hope this can be reduced to smaller bytesize, picking the essential files from folder https://unpkg.com/browse/mermaid@10.0.2/dist/ 

As stated by release notes, the mermaid package mermaid.min.js has been split into multiple packages, just I hope not all 262 files in folder https://unpkg.com/browse/mermaid@10.0.2/dist/ are required.

All in all this folder contains 32MB, in contrast to ca 900KB  of former mermaid.min.js

As the JS files shall be  shipped with the  Rust-Library "aquamarine" the size of 32MB is a bit too large. 

Please tell me, If the following import statement would be used, which further files does it depend on? What is the JS file-set to be shipped as bundle?  Is there a npm-run command  to filter out these files as a bundle using an automated process? 
```
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
```

### Steps to reproduce

Check the content of ttps://unpkg.com/browse/mermaid@10.0.2/dist/ 
For me it not clear which files are required if importing https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs

### Screenshots

_No response_

### Code Sample

_No response_

### Setup

- Mermaid version:  10.0.2
- Browser and Version: rustdoc html build  process embedding mermaid diagrams via https://github.com/mersinvald/aquamarine

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sidharthv96**
## :bookmark_tabs: Summary

This will help downstream users if they want to include only a subset of the dist in their bundles.

image

The files below do not have chunks, hence the missing folders. 
image

Resolves #4203

### CODE REVIEW

**Comment by sidharthv96** [comment]
@aloisklink, what's you opinion on shipping all the `.map` files?
It will significantly reduce the npm install bundle size. Which all .map files are actually needed?

File: .esbuild/util.ts [hunk 1]
```diff
@@ -32,24 +39,31 @@ const buildOptions = (override: BuildOptions): BuildOptions => {
...
-  entryName,
-  metafile,
-  format,
-}: MermaidBuildOptions): BuildOptions => {
+const getFileName = (fileName: string, { core, format, minify }: MermaidBuildOptions) => {
+  if (core) {
+    fileName += '.core';
+  } else if (format === 'esm') {
+    fileName += '.esm';
+  }
```
[the remark this answers is not in the document]
Comment (aloisklink) [reply .esbuild/util.ts:47]: It looks like the current code works fine, but it's just called `chunks/mermaid.core`.

For consistency, maybe it should be the following:

```suggestion
  if (core) {
    fileName += '.core';
  }
  if (format === 'esm') {
    fileName += '.esm';
  }
```

**Review by aloisklink — APPROVED** [review]
> @aloisklink, what's you opinion on shipping all the .map files?
It will significantly reduce the npm install bundle size. Which all .map files are actually needed? (Does core need maps?)

I'd love to have map files, but I think they're currently broken. All of the map files point to the `./src/` dir, but we don't upload the `./src/` dir to NPM, so I don't think they do anything currently.

This is a shame, map files would have helped a lot when I've previously been trying to debug bugs in Mermaid.

My feeling is that:
  - map files are nice to have, but are probably only required by in the `.core.esm.mjs` build, since that's the build that NPM/JavaScript devs are most likely to use.
  - map files need the `src/` dir, but we can't really easily add that to the NPM dir, because the `src/docs` folder has a lot of junk in it (e.g. Vitepress `node_modules` folder??)

So maybe remove the map files for now, and once move the docs out of the `packages/mermaid` folder (see https://github.com/mermaid-js/mermaid/issues/4382), we can consider adding the map files for `core.esm.mjs` back in?

---

As another file saving, we could make both `mermaid.js` and `mermaid.min.js` the same file (same with `mermaid.esm.mjs` and `mermaid.core.esm.mjs`). I know we need to keep both files for backwards compatibility, but I feel like there's no reason why both files can't be the exact same minimized version.

File: .esbuild/util.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sidharthv96) [reply .esbuild/util.ts:47]: core _is_ ESM, the screenshot was from back when the code was how @aloisklink suggested. 
I had to change it to the `if/else` version to keep the existing names. 

image

**Comment by sidharthv96** [comment]
> As another file saving, we could make both mermaid.js and mermaid.min.js the same file (same with mermaid.esm.mjs and mermaid.core.esm.mjs). I know we need to keep both files for backwards compatibility, but I feel like there's no reason why both files can't be the exact same minimized version.

I just realised both are actually the same. But shouldn't mermaid.js be the non minified version?
We are missing a flag in the build options. Setting that fixes the issue.

File: .esbuild/util.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aloisklink) [reply .esbuild/util.ts:47]: > I had to change it to the `if/else` version to keep the existing names.

Good catch! Both me and @Yokozuna59 missed that!

**Review by aloisklink — COMMENTED** [review]
> But shouldn't `mermaid.js` be the non minified version?

It currently should be the non-minified version, yes. But why do we even need to publish a non-minified version? IMO, the only benefit is the non-minified version is easier to debug.

Although, that's probably something to change in a future PR! I'll say keep the old behavior (`mermaid.js` and `mermaid.esm.mjs` are non-minified) now. We can then make another PR in the future (maybe even for a Mermaid v11.1 release), that will minify them to save space. That PR will probably need to get a :+1: from more reviewers, since although I think it should be a non-breaking change, touching the entry-points is always high risk.

**Pull request merged**